src/test/scala/com/databricks/spark/avro/AvroFileGenerator.scala

```diff
@@ -0,0 +1,96 @@
+package com.databricks.spark.avro
+
+import java.io.File
+import java.util.HashMap
+import java.util.Vector
+import java.nio.ByteBuffer
+
+import scala.util.Random
+
+import org.apache.avro._
+import org.apache.avro.file.DataFileWriter
+import org.apache.avro.generic._
+
+/**
+ * This object allows you to generate large avro files that can be used for speed benchmarking.
+ * See README on how to use it.
+ */
+object AvroFileGenerator {
+  
+  val defaultNumberOfRecords = 1000000
+  val defaultNumberOfFiles = 1
+  val outputDir = "src/test/resources/avroForBenchmark/"
+  val schemaPath = "src/test/resources/benchmarkSchema.avsc"
+  val objectSize = 100 // Maps, arrays and strings in our generated file have this size
+
+  def generateRandomMap(rand: Random): java.util.Map[String, Int] = {
+    val jMap = new HashMap[String, Int]()
+    for (i <- 0 until objectSize) {
+      jMap.put(rand.nextString(5), i)
+    }
+    jMap
+  }
+  
+  def generateRandomArray(rand: Random): Vector[Boolean] = {
```

Comment:
Any particular reason you used util.Vector? util.ArrayList is more common unless you explicitly need the synchronization advantages of the former.